pretty_midi/pretty_midi.py

```diff
@@ -402,6 +403,161 @@ def estimate_tempo(self):
         """
         return self.estimate_tempi()[0][0]
 
+    def get_beats_using_metadata(self):
+        """Uses Time Signature and Tempo metadata to estimate beat times
+
+        Returns
+        -------
+        beat_times : np.ndarray
+            List of beat times in seconds
+        """
+
+        # threshold in seconds for merging beat locations
+        epsilon = 0.05
+
+        # get qpms (quarter per minute) tempi and their change time
+        tempo_change_times, qpms = self.get_tempo_changes()
+
+        # container for storing beat times
+        beat_times = []
+
+        # store time signature objects
+        ts_idx = 1
+        if len(self.time_signature_changes) > 1:
+            cur_ts = self.time_signature_changes[0]
+            nxt_ts = self.time_signature_changes[1]
+        else:
+            cur_ts = self.time_signature_changes[0]
+            nxt_ts = None
+
+        # go over tempo change times and time signatures assuming that the first
+        # tempo change time is the start time of the first beat
+        i = 1
+        beat_start = tempo_change_times[0]
+        while i < len(tempo_change_times):
+            # create beat times by going over time signatures while the next
+            # time signature happens before the next tempo change. note that
+            # time signatures always begin on the start of a beat
+            while nxt_ts is not None and nxt_ts.time < tempo_change_times[i]:
+                # update bpm given current time signature and qpm
+                bpm = qpm_to_bpm(qpms[i-1],
+                                 cur_ts.numerator,
+                                 cur_ts.denominator)
+
+                # interpolate from cur to nxt time signature change time
+                beat_times = np.hstack((beat_times, np.arange(beat_start,
```

Comment:
What do you mean?  `np.linspace` and `np.arange` can be used equivalently, always, except that linspace is never wrong.  If you know the start, stop, and step size, as you do here:

```Python
In [1]: start = 0.2
In [2]: stop = 2.2
In [3]: step = .1
In [4]: np.arange(start, stop + step, step)
Out[4]:
array([ 0.2,  0.3,  0.4,  0.5,  0.6,  0.7,  0.8,  0.9,  1. ,  1.1,  1.2,
        1.3,  1.4,  1.5,  1.6,  1.7,  1.8,  1.9,  2. ,  2.1,  2.2]

In [5]: np.linspace(start, stop, (stop - start)/step + 1)
Out[5]:
array([ 0.2,  0.3,  0.4,  0.5,  0.6,  0.7,  0.8,  0.9,  1. ,  1.1,  1.2,
        1.3,  1.4,  1.5,  1.6,  1.7,  1.8,  1.9,  2. ,  2.1,  2.2])
```

I'd prefer this a lot more than to allowing `np.arange` to potentially make unpredictable mistakes and then trying to fix them later.  That allows for a lot of edge cases that we may or may not account for.